src/SourceLocator/Type/Composer/Factory/MakeLocatorForInstalledJson.php

```diff
@@ -173,9 +175,9 @@ private function packageToFilePaths(array $package): array
     /**
      * @param array{name: string, autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}} $package
      */
-    private function packagePrefixPath(string $trimmedInstallationPath, array $package): string
+    private function packagePrefixPath(string $trimmedInstallationPath, array $package, string $vendorDir): string
     {
-        return $trimmedInstallationPath . '/vendor/' . $package['name'] . '/';
+        return $trimmedInstallationPath . '/' . $vendorDir . '/' . $package['name'] . '/';
```

Comment:
Can `$vendorDir` include leading or trailing `/`?

Reply:
Will check. But valid question :)

Reply:
A tailing `/` in the composer.json config, breaks that test, so I'll look into this.

Reply:
I would need to sanitize the `$vendorDir` before handing it over. A `ltrim($vendorDir, "/")` and `rtrim($vendorDir, "/")` could be an option, but not really sure how I like it, and where to handel this. 

Do you have any input on this? 

I'll add a test case with leading and tailing `/` to catch this if anything changes. 
```json
{
  "config": {
    "vendor-dir": "/custom-vendor/"
  }
}
```

Reply:
I think it's only `rtrim()`, since `/` means "absolute path", such as `/tmp/vendor`

Reply:
True. But the code base doesn't read it as an absolute path, This is the errors from the config example above. So if someone sets it with both leading and tailing, it has to be solved, or is that out of scope and one should know?

```shell
0 => '/Projects/php/roave/BetterReflection/test/unit/Assets/ComposerLocators/project-with-custom-vendor-dir-using-composer-v2//custom-vendor//a/b/src/ab_PSR-4_Sources'
```

Reply:
> or is that out of scope and one should know?

I'd say out of scope => new ticket for later. Using an absolute path for `vendor` is very harmful, and certainly goes against all principles on which composer is based upon. In addition to that, tools like `roave/backward-compatibility-check` wouldn't even work then (due to path collisions)

Reply:
I don't know if we are misunderstanding each other? 

Composer doesn't see the `/` in config: vendor-dir as a absolute path, it replaces the "vendor" in the full path. 

e.g. 
`/path/to/project/vendor/....` will be ``/path/to/project/custom-vendor/....`

The question would be if composer validation should handle that as invalid configuration. To be honest I don't know how much validation is in the composer.json schema.

Reply:
That's alright, hence why I propose to defer handling of a leading `/`, and just adjust things with `rtrim()` only